hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -688,14 +688,23 @@ case class TableMultiWayZipJoin(children: IndexedSeq[TableIR], fieldName: String
     }
 
     val childRVDs = childValues.map(_.rvd)
-    val childRanges = childRVDs.flatMap(_.partitioner.rangeBounds)
-    val newPartitioner = RVDPartitioner.generate(childRVDs.head.typ.kType.virtualType, childRanges)
-    val repartitionedRVDs = childRVDs.map(_.repartition(newPartitioner))
+    val repartitionedRVDs =
+      if (childRVDs.forall(rvd =>
+        (rvd.partitioner.rangeBounds sameElements childRVDs(0).partitioner.rangeBounds) &&
+          rvd.partitioner.allowedOverlap == 0))
```

Comment:
I'm going to zip together a bunch of different inputs.  So, if the partitions overlap, it may be some keys in this partition are in another, overlapping partition in another input, no?  So I want to check there is no overlap.